hack/cnbi-package-list.yaml

```
@@ -0,0 +1,199 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: cnbi-package-list
+  annotations:
+    meteor.zone/pipeline: "true"
+    shower.meteor.zone/button-label: Run code in JupyterHub
+spec:
+  description: Build a notebook image from a package list
+  params:
+    - name: baseImage
+      description: Container image repository url
+      type: string
+    - name: packages
+      type: array
+    - name: name
+      description: Image name
+      type: string
+    - name: description
+      description: Custom description
+      type: string
+    - name: creator
+      description: Owner, user who requested the import
+      type: string
+  # TODO: baseImage selection with osVersion, osName, pythonVersion
+  workspaces:
+    - name: packages-to-install
+
+  tasks:
+    - name: resolve-dependencies
+      workspaces:
+        - name: pinned-packages
+          workspace: packages-to-install
+      taskRef:
+        name: get-package-versions-list
+      params:
+        - name: PACKAGES
+          value:
+            - $(params.packages)
+    ####### TODO: the rest of the tasks sould be shared between all types of sources
+    - name: get-base-image
+      params:
+        - name: baseImage
+          value: $(params.baseImage)
+      taskSpec:
+        params:
+          - name: baseImage
+            type: string
+        results:
+          - name: baseImage
+        steps:
+          - script: | # PLACEHOLDER
+              image=quay.io/thoth-station/s2i-custom-notebook
+              if [ -n "$(params.baseImage)" ]
+              then
+                image="$(params.baseImage)"
+              fi
+              echo $image > $(results.baseImage.path)
+            image: alpine
```

I am running into dockerhub rate limiting issues when testing this.

Maybe this alternative could work here:
```suggestion
            image: registry.access.redhat.com/ubi9-micro
```

---

Good idea, I'll use that (unless we have transparent caching repository for docker hub ?)